### PR TITLE
Add current workspace member option to workspaceMember relation field filter (#8016) 

### PR DESCRIPTION
**Pull request by eliasylonen**
Issue: #8016

Almost ready to be merged.

This touches so many files that it could be good to:
- merge quickly after I fix the tests
- Add the "No Owner" / "No Assignee" button mentioned in #8016 separately
- Fix unselecting values not working (also broken in main) right after merging

### CODE REVIEW

**Comment by FelixMalfait** [comment]
Great work as always! Starting to play around with it, noticed a first bug: 

https://github.com/user-attachments/assets/7f3d2d39-d9f6-4f6d-90b7-70cebc3af8e6

**Comment by FelixMalfait** [comment]
In a followup PR / issue, could you maybe add an "assigned to me" view in "tasks-all.view.ts" file (this will only apply to new workspaces). This will make your feature more discoverable

**Comment by FelixMalfait** [comment]
I think we can forget the "No assignee" from the initial issue as there is already "is empty which is good enough"

**Comment by FelixMalfait** [comment]
Maybe create a followup issue to add it on the other side of the relationship?

Screenshot 2024-12-09 at 10 23 17

**Comment by lucasbordeau** [comment]
Please fix the tests

**Comment by lucasbordeau** [comment]
Could you also verify that your IDE is correctly configured ? The lint problem on the tests and the prettier/eslint problems should not be part of your commit since it should be auto-fixed at save.

File: packages/twenty-front/src/modules/views/view-filter-value/utils/resolveBooleanViewFilterValue.ts [hunk 1]
```diff
@@ -1,7 +1,2 @@
-import { ViewFilter } from '@/views/types/ViewFilter';
-
-export const resolveBooleanViewFilterValue = (
-  viewFilter: Pick<ViewFilter, 'value'>,
-) => {
-  return viewFilter.value === 'true';
-};
+export const resolveBooleanViewFilterValue = (viewFilterValue: string) =>
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/views/view-filter-value/utils/resolveBooleanViewFilterValue.ts:1]: We can keep the previous function, we don't strive to compact the code in this codebase, it's okay to keep things explicit and less compact.

File: packages/twenty-front/src/modules/views/view-filter-value/utils/resolveDateViewFilterValue.ts [hunk 2]
```diff
@@ -159,32 +158,28 @@ const resolveVariableDateViewFilterValueFromRelativeDate = (relativeDate: {
...
-    ? ReturnType<typeof resolveVariableDateViewFilterValue>
+    ? ResolvedRelativeDateFilterValue
     : Date | null;
 
-type PartialViewFilter<O extends ViewFilterOperand> = Pick<
-  ViewFilter,
-  'value'
-> & { operand: O };
-
 export const resolveDateViewFilterValue = <O extends ViewFilterOperand>(
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/views/view-filter-value/utils/resolveDateViewFilterValue.ts:175]: Is this refactor needed for this feature ?

File: packages/twenty-front/src/modules/views/hooks/useResolveFilterValue.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        ? boolean
+        : T extends 'RELATION'
+          ? string[]
+          : string;
+
+// TODO: Convert all resolve functions into hooks
+export const useResolveFilterValue = () => {
+  const { resolveRelationViewFilterValue } = useResolveRelationViewFilterValue();
+
+  const resolveFilterValue = <
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/views/hooks/useResolveFilterValue.ts:31]: We should keep this as a util

File: packages/twenty-front/src/modules/views/view-filter-value/utils/resolveNumberViewFilterValue.ts [hunk 4]
```diff
@@ -1,7 +1,6 @@
-import { ViewFilter } from '@/views/types/ViewFilter';
 
 export const resolveNumberViewFilterValue = (
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/views/view-filter-value/utils/resolveNumberViewFilterValue.ts:2]: Is this refactor needed ?

File: packages/twenty-front/src/utils/validation-schemas/safeStringArrayJSONSchema.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { z } from 'zod';
+
+export const safeStringArrayJSONSchema = z
```
Comment (lucasbordeau) [inline packages/twenty-front/src/utils/validation-schemas/safeStringArrayJSONSchema.ts:3]: Ok to extract this as it was but the validation seems to introduce a silent error that returns [], could we throw or  return a wrong validation state instead when the value is really not parsable as an array ?

**Review by lucasbordeau — CHANGES_REQUESTED** [review]
With selectedRecordIdsAndSpecialIds, the naming tells us that there's a code smell, "DoingThisAndThat" indicates that we should split this into two distinc flows that join at the right abstraction level.

Here we should keep the previous flow as it is, with utils instead of hooks.

The changes seem to be only for relations and selection. You should explore a solution where you create recoil atoms for storing the state of what your new UI components are doing, then using those atoms where it's really needed, and copy/paste at multiple place if that's the things to do.

For example here contextStoreFilters seems to be more appropriate to store your additional current workspace member filter, and maybe a specific parallel hook/util could handle turning the click into a record filter and modify this state ?

**Comment by eliasylonen** [comment]
Continuing in PR #9182

**Pull request closed without merging**